include/mqtt/client.hpp

```diff
@@ -1434,6 +1523,8 @@ class client : public endpoint<std::mutex, std::lock_guard, PacketIdBytes> {
 #if defined(MQTT_USE_WS)
     std::string path_;
 #endif // defined(MQTT_USE_WS)
+    std::uint32_t session_expiry_interval_ = 0;
+    as::steady_timer tim_session_expiry_;
```

Comment:
Does client side need to have timer at all?

I thought the timer was a server-side only thing?

Reply:
At first, I thought so too.
However, the sped said that
https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901048

> The Client and Server MUST store the Session State after the Network Connection is closed if the Session Expiry Interval is greater than 0 [MQTT-3.1.2-23].


https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901231

> The Client and Server MUST NOT discard the Session State while the Network Connection is open [MQTT-4.1.0-1]. The Server MUST discard the Session State when the Network Connection is closed and the Session Expiry Interval has passed [MQTT-4.1.0-2].

Consider the following scenario:

1. A client CONNECT (SessionExpiryInterval=10) to the broker. Note CleanStart is not important here.
2. The client PUBLISH QoS1 message1 to topic1 PacketId=100
3. Network is disconnected before receiving PUBACK PacketId=100.
4. Wait some seconds.
5. The client CONNECT (CleanStart=false) again to the broker. Note SessionExpiryInterval is not important here.

If waiting seconds< 10, the client should resend PUBLISH at the step2.
If waiting second >= 10, the client shouldn't resend PUBLISH at the step2. 

The difference is caused by the client side session expiration process.

Reply:
Ok.